soh/soh/OTRGlobals.cpp

```diff
@@ -1509,3 +1453,89 @@ extern "C" s32 Randomizer_GetRandomizedItemId(GetItemID ogId, s16 actorId, s16 a
 extern "C" s32 Randomizer_GetItemIdFromKnownCheck(RandomizerCheck randomizerCheck, GetItemID ogId) {
     return OTRGlobals::Instance->gRandomizer->GetRandomizedItemIdFromKnownCheck(randomizerCheck, ogId);
 }
+
+extern "C" CustomMessageEntry Randomizer_GetCustomGetItemMessage(GetItemID giid, char* buffer, const int maxBufferSize) {
+    const CustomMessageEntry getItemText = CustomMessageManager::Instance->RetrieveMessage(Randomizer::getItemMessageTableID, giid);
+    return getItemText;
+}
+
+extern "C" int CustomMessage_RetrieveIfExists(GlobalContext* globalCtx) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    uint16_t textId = msgCtx->textId;
+    Font* font = &msgCtx->font;
+    char* buffer = font->msgBuf;
+    const int maxBufferSize = sizeof(font->msgBuf);
+    CustomMessageEntry messageEntry;
+    if (gSaveContext.n64ddFlag) {
+        if (textId == TEXT_RANDOMIZER_GI_ESCAPE_HATCH) {
```

Comment:
`TEXT_RANDOMIZER_GI_ESCAPE_HATCH` is a really fun name, but it'd be nice to try to come up with something a little more descriptive. i'm not sure what a better name would be but figured it's worth noting that it was unclear to me when reading what it is supposed to be/do

Reply:
I thought it was pretty descriptive, as it describes the text id used to "escape" from the usual code for getting messages during get-item scenario into the custom text system. Would `TEXT_RANDOMIZER_CUSTOM_ITEM` be a better choice?